src/engraving/layout/layoutsystem.cpp

```diff
@@ -1505,3 +1510,102 @@ void LayoutSystem::restoreTies(System* system)
     Fraction etick = system->measures().back()->endTick();
     doLayoutTies(system, segList, stick, etick);
 }
+
+void LayoutSystem::manageNarrowSpacing(System* system, double& curSysWidth, double targetSysWidth)
```

Comment:
This algorithm looks very suboptimal. We iterate over this system->measures() list many many times (including some implicit iterations, such as inside system->minSysTicks/maxSysTicks). Considering how slow the layout process is, we should at least do some performance tests before merging this